### ISSUE TITLE
Object editor changes - Oracle

### ISSUE THREAD

**Issue by premyslc** [dbeaver/dbeaver#6905]
1. When I open some source and i change something by mistake, dbeaver
ask me to save changes to DB. It's situation near big trouble. ;-)
2. Oracle - I see object error (package, procedure...) only after compilation.

**Describe the solution you'd like**
1. Allow to open sources in read only mode by default. Some general setting for editor? Then allow editing by some button in object editor (readonly switch).
2. Load actual errors for source from dba_errors / all_errors.

**Comment by p91paul** [comment]
for 2, which DBeaver version are you using? 6.2.1 should include a fix for this.

**Comment by premyslc** [comment]
I'm using 6.2.0, then 6.2.1 will be fine. Thanks. :-)

**Comment by premyslc** [comment]
Seem (2) it doesnt work in 6.2.4. I see only information about object validity and no errors.

**Comment by p91paul** [comment]
yeah, I misunderstood your question. Since 6.2.1 there was a fix for packages compilation on save.

When editor is opened you don't have the error list available if you don't compile (which could be impossible if you're not the procedure/package owner), so your request would be a nice addition! I hope the developers pick this up
